gunicorn.conf.py

```diff
@@ -1,22 +1,31 @@
+import multiprocessing
 from psycogreen.gevent import patch_psycopg
 import os
 DEPLOYMENT_MODE = os.environ.get('DEPLOYMENT_MODE')
 
 worker_class = 'gevent'
-workers = 4
-# to stay under heroku limit of 20 connections
+workers = 2
+# to stay under heroku limit of 20 connections,
+# but with these should get caught by the pool
 worker_connections = 4
-
+preload = True
 pythonpath = "/app/joplin"
-reload = True
+
 #  see https://docs.gunicorn.org/en/stable/settings.html#max-requests
 max_requests = 500
 max_requests_jitter = 50
 timeout = 190
-preload = True
+
+if DEPLOYMENT_MODE in ("STAGING", "PRODUCTION"):
+    timeout = 190
+    worker_connections = 10
+    workers = multiprocessing.cpu_count() * 2
+
+
 if DEPLOYMENT_MODE in ("LOCAL", "REVIEW"):
     timeout = 190
     loglevel = "DEBUG"
+    reload = True
```

Comment:
we shouldn't have reload on in production, and staging should mirror production closely